activators/dialogflow/src/main/kotlin/com/justai/jaicf/activator/dialogflow/DialogflowIntentActivator.kt

```diff
@@ -44,9 +39,47 @@ class DialogflowIntentActivator(
         return intent?.let { listOf(DialogflowActivatorContext(intent, qr)) } ?: emptyList()
     }
 
-    class Factory(private val connector: DialogflowConnector): ActivatorFactory {
+    override fun fillSlots(
+        request: BotRequest,
+        reactions: Reactions,
+        botContext: BotContext,
+        activatorContext: ActivatorContext?,
+        slotReactor: SlotReactor?
+    ): SlotFillingResult {
+        var context = activatorContext?.dialogflow?.apply {
+            if (queryResult.allRequiredParamsPresent) {
+                return SlotFillingSkipped
+            }
+        }
+        if (context == null) {
+            context = recogniseIntent(botContext, request).firstOrNull()
+        }
+        return when {
+            context == null -> SlotFillingInterrupted
```

Comment:
It seems like if an intent has required parameters, then on the first "interruption" by user, dialogflow will interrupt slotfilling, but will return the same intent in the response. So, in the current implementation, we will return `SlotFillingInProgress` in this case, as I understand

User: Hello
Dialogflow: "HelloIntent", action = "user_defined_slotfilling_action_name"
Bot: Hi, what is your name? (required) 
User: Cancel
Dialogflow: "HelloIntent", action = null
User: Cancel
Dialogflow: "Fallback"

===================================

Also, we should check `queryResult.isFallback`, because even in the case of fallback Dialogflow will return intent, and context will not be null. I guess, this check should be done in `recognizeIntent`

Reply:
No, in this case we return `SlotFillingFinished`. You can just check the current example using "wake up" input to trigger the slot filling and then send "cancel".